src/Cake.Core/Polyfill/EnvironmentHelper.cs

```diff
@@ -119,7 +120,10 @@ public static Runtime GetRuntime()
         public static FrameworkName GetBuiltFramework()
         {
 #if NETCORE
-            return new FrameworkName(".NETStandard,Version=v2.0");
+            return new FrameworkName(Assembly
+                .GetEntryAssembly()?
+                .GetCustomAttribute<TargetFrameworkAttribute>()?
+                .FrameworkName ?? ".NETStandard,Version=v2.0");
```

Comment:
This allows us to get `.NETCoreApp,Version=v2.1` for `Cake.Tool` and `.NETCoreApp,Version=v2.0` for `Cake.CoreCLR`.